main.rb

```diff
@@ -118,149 +129,27 @@ def my_map(is_proc = nil)
   end
 
   def my_inject(temp_option = nil)
-    new_items = temp_option || shift
-    my_each do |item|
-      new_items = yield(new_items, item)
+    if is_a? Range
+      min = self.min
+      max = self.max
+      new_items = temp_option || 0
+      while min <= max
+        new_items = yield(new_items, min)
+        min += 1
+      end
+    else
+      new_items = temp_option || shift
+      my_each do |item|
+        new_items = yield(new_items, item)
+      end
     end
     new_items
   end
 end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
array.my_inject(&operation) = array.inject(&operation)
```
Should return `true`